clients/drcachesim/tracer/tracer.cpp

```diff
@@ -413,13 +422,29 @@ memtrace(void *drcontext, bool skip_size_cap)
                                               dr_get_thread_id(drcontext));
     pipe_start = data->buf_base;
     pipe_end = pipe_start;
-    if (!skip_size_cap && op_max_trace_size.get_value() > 0 &&
-        data->bytes_written > op_max_trace_size.get_value()) {
+    if (!skip_size_cap &&
+        (data->output_disabled ||
```

Comment:
Do we really need the `data->output_disabled` condition? As we already have `is_beyond_global_max` here.

Reply:
The counter could easily overflow otherwise, and DR doesn't provide an atomic that sets a value and reads the old value (it has that internally: could export one).  A `std::atomic<bool>` would be much simpler: maybe we should go with that and assume there will be no isolation problem.

Reply:
Went with atomic_load per thread below.